Purchases/Purchasing/TransactionsFactory.swift

```diff
@@ -14,17 +14,41 @@
 
 import Foundation
 
-class TransactionsFactory {
-
-    func nonSubscriptionTransactions(withSubscriptionsData subscriptionsData: [String: [[String: Any]]],
-                                     dateFormatter: DateFormatterType) -> [Transaction] {
-        subscriptionsData.flatMap { (productId: String, transactionData: [[String: Any]]) -> [Transaction] in
-            transactionData.map {
-                Transaction(with: $0, productId: productId, dateFormatter: dateFormatter)
-            }.compactMap { $0 }
-        }.sorted {
-            $0.purchaseDate < $1.purchaseDate
-        }
+enum TransactionsFactory {
```

Comment:
the one advantage of class over enum for these is that they're easier to DI for tests

Reply:
then again, this is private and it looks like we're not testing its usage yet

Reply:
Yeah we don't mock it, and it's the kind of function that can't really have multiple possible implementations.